.github/workflows/test-firedrake.yml

```diff
@@ -17,59 +17,87 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 60
     container:
-      image: firedrakeproject/firedrake:latest
+      image: firedrakeproject/firedrake-vanilla:latest
       options: --user root
     env:
       OMP_NUM_THREADS: 1
       OPENBLAS_NUM_THREADS: 1
     steps:
-      - name: Checkout 
+      - name: Checkout
         uses: actions/checkout@v4
         with:
           path: 'tlm_adjoint'
       - name: Install dependencies
         run: |
-          . /home/firedrake/firedrake/bin/activate
-          python3 -m pip install jax[cpu] ruff pytest-timeout pytest-xdist
+          cp -pr ${GITHUB_WORKSPACE}/tlm_adjoint /home/firedrake
+          chown -R firedrake:firedrake /home/firedrake/tlm_adjoint
+          apt-get update
+          apt-get install -y flake8
+          su - firedrake
+          export HOME=/home/firedrake
```

Comment:
If this works then that's great. I couldn't get `su firedrake` to work so my approach was to set `PYTHONPATH` which is way worse.